src/NuGet.Core/NuGet.Commands/RestoreCommand/Logging/TransitiveNoWarnUtils.cs

```diff
@@ -902,18 +934,23 @@ private static HashSet<NuGetLogCode> Intersect(HashSet<NuGetLogCode> first, Hash
                     return first;
                 }
 
-                if (first == null)
+                // null is empty and cannot intersect
+                if (first == null || first.Count == 0)
                 {
-                    return second;
+                    return null;
```

Comment:
Are we breaking any edge case previously working but now it doesn't work anymore?
I'm not familiar with this part of the code, so can't think of one for now.

Reply:
I couldn't find any edge-case that was broken.  But I'll admit that I'm coming to this trying to tackle a performance blocker and am not an expert on NuGet warnings.